Plane: remove TRIM_AUTO and trim_radio pre-flight calibration

IamPete1 opened the pull request:
A step on the way to https://github.com/ArduPilot/ardupilot/pull/18514 

I went to remove `trim_radio` preflight calibration and found there is a param to do it on exit from manual mode also. I have removed both. As the `TRIM_AUTO` param description said "See newer and safer SERVO_AUTO_TRIM parameter for automated results."

peterbarker commented on ArduPlane/Parameters.h:
```diff
@@ -51,7 +51,7 @@ class Parameters {
         // Misc
         //
-        k_param_auto_trim      = 10,
+        k_param_auto_trim      = 10, // unused
```
```suggestion
        k_param_auto_trim_unused      = 10, // unused
```

IamPete1 replied:
There are lots without this, including the ones I recently removed....

peterbarker reviewed (approved):
I like it.

Hwurzburg commented:
not sure, but doesnt this break a feature in QGC where a user can press a button to save current trims, then reset to center on TX?

IamPete1 commented:
Well, if it does work now were using the wrong param in the mavlink message. The GCS does not really need a message to do this anyway, it knows what all the trims and servos are doing so it can just set the params like it does for normal RC calibration.

peterbarker commented:
@Hwurzburg I haven't tested, but it looks like QGC uses the standard value rather than the incorrect one we're using: https://github.com/mavlink/qgroundcontrol/blob/master/src/Vehicle/Vehicle.cc#L3251

Hwurzburg commented:
Probably why it did not work for me in QGC when I tried it....probably should fix and retain then

tridge reviewed (changes requested):
should use a RCn_OPTION instead, which adjusts RC trims and SERVOn_TRIM only when in MANUAL mode
change margin to 30%, and check roll and pitch rates below 10%

IamPete1 commented:
Added `TRIM_TO_CURRENT_SERVO_RC` RC option. I have moved rover over to it from save trim. Save trim is uses on copter for AHRS trim, so I think it is better that rover and plane who are doing servo and rc trim both move to the new option. 

I have removed `TRIM_AUTO` and the mavlink trigger from plane.

I have then added the manual mode and rotation rate checks to the existing plane `trim_radio` function along with useful messages.

tridge commented on ArduPlane/radio.cpp:
```diff
@@ -299,24 +299,32 @@ void Plane::control_failsafe()
+        return;
+    }
+
+    if (degrees(ahrs.get_gyro().length()) > 5.0) {
```
change threshold to 30 deg/sec

IamPete1 replied:
done

tridge reviewed (changes requested):
just need to change the threshold

IamPete1 commented:
rebased
